samples/snippets/csharp/VS_Snippets_CLR/conceptual.disposable/cs/using3.cs

```diff
@@ -2,27 +2,31 @@
 using System;
 using System.IO;
 
-public class Example
+class Example
 {
-   public static void Main()
-   {
-      Char[] buffer = new Char[50];
-      {
-         StreamReader s = new StreamReader("File1.txt");
-         try {
-            int charsRead = 0;
-            while (s.Peek() != -1) {
-               charsRead = s.Read(buffer, 0, buffer.Length);
-               //
-               // Process characters read.
-               //
+    static void Main()
+    {
+        var buffer = new char[50];
+
+        var streamReader = new StreamReader("file1.txt");
+        try
+        {
+            var charsRead = 0;
+            while (streamReader.Peek() != -1)
+            {
+                charsRead = streamReader.Read(buffer, 0, buffer.Length);
+                //
+                // Process characters read.
+                //
+            }
+        }
+        finally
+        {
+            if (streamReader != null)
+            {
+                ((IDisposable)streamReader).Dispose();
```

Comment:
I don't think the casting is required here, in StreamReader's hierarchy IDisposable isn't implemented explicitly.

So the finally can just use `streamReader?.Dispose();`

Or just `streamReader.Dispose();` if it's confident that no one is going to null-out the variable in `// Process characters read.`

Reply:
In this example we were demonstrating that intentionally.

> When the C# or Visual Basic compiler encounters the `using` statement, it emits intermediate language (IL) that is equivalent to the following code that explicitly contains a `try/finally` block.

Reply:
Ah, my bad.